packages/main/src/queue/index.ts

```diff
@@ -248,36 +248,54 @@ const handlers = {
     }
   }),
 
-  flag: handler({
-    enqueue(params: { accountId: ID; box: { name: string }; uids: number[] }) {
-      cache.addFlag({ ...params, flag: "\\Flagged" })
+  setFlagged: handler({
+    enqueue(params: {
+      accountId: ID
+      box: { name: string }
+      uids: number[]
+      isFlagged: boolean
+    }) {
+      params.isFlagged
+        ? cache.delFlags({ ...params, flags: ["\\Flagged"] })
+        : cache.addFlag({ ...params, flag: "\\Flagged" })
```

Comment:
This is still backward. If `isFlagged` is `true` I think that means that we are instructing the service to make the conversation flagged. Don't think of `isFlagged` as the prior state of the conversation, think of it as the end result.